servant/src/Servant/Utils/Enter.hs

```diff
@@ -39,6 +39,9 @@ instance ( Enter typ1 arg1 ret1, Enter typ2 arg2 ret2
 instance (Enter b arg ret) => Enter (a -> b) arg (a -> ret) where
     enter arg f a = enter arg (f a)
 
+instance Enter (arg -> ret) arg ret where
+    enter arg f = f arg
```

Comment:
It feels a bit arbitrary to have an instance like this. Maybe it's better to have some newtype wrapper for the argument?

``` haskell
instance Enter (arg -> ret) (InjectedArgument arg) ret where
  enter (InjectedArgument arg) f = f arg
```